src/services/api/variant/index.js

```diff
@@ -180,6 +181,29 @@ export default class VariantService extends ApiService {
 
         } ) )
 
+        // Generate excel file for list of variants
+        this.instance.post( {
+          path: `${this.config.endpoint}/xl`,
+        }, async (req, res, next) => {
```

Comment:
I find wrapping sendDataAsExcel in an endpoint handler makes things more convoluted then they have to be.

I understand the desired to decouple error login from the endpoint handler though you do have some choices here:
- Make  sendDataAsExcel a higher order function and either pass the logger or even complete error handler to it as an argument. An example of this pattern can be found here (also, you might take the opportunity to log your error message in json format which will make it easier to centralize later): https://github.com/cr-ste-justine/score-auth/blob/master/src/middleware/access_control.js#L21
- Put a separate error handler that logs the message (you can do it with Express and I believe Restify is pretty similar in behavior)

Reply:
After looking at the remainder of the code, I see that it follows a similar pattern.

While I think the implementation adds extra complexity without giving value in return, I will not force my opinion on a single new endpoint without at least refactoring other endpoints in the same file which is out of scope for this PR I so I'll close this discussion and leave it be.